app/src/main/java/com/example/simbirsoft_android_practice/filter/FilterPreferenceDataStore.kt

```diff
@@ -0,0 +1,41 @@
+package com.example.simbirsoft_android_practice.filter
+
+import android.content.Context
+import androidx.datastore.preferences.core.PreferenceDataStoreFactory
+import androidx.datastore.preferences.core.edit
+import androidx.datastore.preferences.core.stringSetPreferencesKey
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.map
+import java.io.File
+import javax.inject.Inject
+
+private const val PREFS_NAME = "filter_prefs.preferences_pb"
+private const val KEY_SELECTED_CATEGORIES = "selected_categories"
+
+class FilterPreferenceDataStore @Inject constructor(
+    context: Context
+) {
+
+    private val dataStore = PreferenceDataStoreFactory.create(
+        produceFile = {
+            File(context.filesDir, PREFS_NAME)
+        }
+    )
```

Comment:
Очень грубо создаёшь инстанс File. Можно сделать проще
```suggestion
val Context.dataStore: DataStore<Preferences> by preferencesDataStore(name = PREFS_NAME)

class FilterPreferenceDataStore @Inject constructor(
    context: Context
) {

    private val dataStore = context.dataStore
```